packages/web/src/services/fingerprint/FingerprintClient.ts

```diff
@@ -0,0 +1,74 @@
+import FingerprintJS, { Agent } from '@fingerprintjs/fingerprintjs-pro'
+
+import { isElectron, isMobile } from 'utils/clientUtil'
+
+const IDENTITY_SERVICE = process.env.REACT_APP_IDENTITY_SERVICE
+
+class FingerprintClient {
+  private apiKey: string
+  private fingerprint: Agent | null
+  private endpoint: string
+
+  constructor(apiKey: string, endpoint: string) {
+    this.apiKey = apiKey
+    this.fingerprint = null
+    this.endpoint = endpoint
+  }
+
+  async init() {
+    console.log('Initializing Fingerprint client')
+    try {
+      const fp = await FingerprintJS.load({
+        apiKey: this.apiKey,
+        endpoint: this.endpoint
+      })
+      console.log(`Fingerprint loaded`)
+      this.fingerprint = fp
+    } catch (e) {
+      console.error(`Error initializaing fingerprint client: ${e}`)
+    }
+  }
+
+  async identify(userId: number) {
+    if (!this.fingerprint) {
+      console.warn('Fingerprint client not yet initted')
+      return
+    }
+    try {
+      const origin = isMobile() ? 'mobile' : isElectron() ? 'desktop' : 'web'
+
+      // First, see if we've fingerprinted this user before
+      const response = await fetch(
```

Comment:
Non-blocking, but we could probably avoid an extra network req if we remember in local storage that we have done a fprint

Reply:
interesting idea - would be a bit problematic in the case that you switch browsers / use incognito, and we'd end up fingerprinting multiple times. going to leave as if for now

Reply:
don't we want to fp again if you switch browsers?

Reply:
discussed online – fp is stable between browsers